src/liquid/components/ld-sidenav/ld-sidenav.tsx

```diff
@@ -448,7 +449,20 @@ export class LdSidenav {
     this.collapsed = !this.collapsed
   }
 
-  private activeSubnavContainsIcons = () => {
+  private activeSubnavContainsIconsOrHasBack = () => {
+    // Return true if we have a back button or a nav items as a direct child.
+    if (
+      Array.from(this.el.children).filter(
+        (child) =>
+          child.tagName === 'LD-SIDENAV-BACK' ||
+          (child.tagName === 'LD-SIDENAV-NAVITEM' &&
+            !(child as HTMLLdSidenavNavitemElement).mode)
+      ).length
+    ) {
+      return true
+    }
```

Comment:
This whole filter expression is kind of hard to read especially directly as if condition.
Additionally it's kind of unfortunate that mode defaults to udefined although it is "basically" an enum which is omitting `primary` as the default value.

Anyways, I think to improve readability, how about extracting the filter to a variable.
```suggestion
    const backAndPrimaryNavItems = Array.from(this.el.children).filter(
        (child) =>
          child.tagName === 'LD-SIDENAV-BACK' ||
          (child.tagName === 'LD-SIDENAV-NAVITEM' &&
            !(child as HTMLLdSidenavNavitemElement).mode)
      )
    if (backAndPrimaryNavItems.length > 0){
      return true
    }
```

Sorry formatting does not work out well.